src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -36,6 +37,37 @@ public class PackageArchiveReader : PackageReaderBase
         /// </summary>
         protected Stream ZipReadStream { get; set; }
 
+        /// <summary>
+        /// Nupkg package reader
+        /// </summary>
+        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing.</param>
+        /// <param name="compatibilityProvider">Framework compatibility provider.</param>
+        private PackageArchiveReader(IFrameworkNameProvider frameworkProvider, IFrameworkCompatibilityProvider compatibilityProvider)
+            : base(frameworkProvider, compatibilityProvider)
+        {
+            _environmentVariableReader = EnvironmentVariableWrapper.Instance;
+        }
+
+        // For testing purposes only
+        internal PackageArchiveReader(Stream stream, IEnvironmentVariableReader environmentVariableReader)
+            : this(stream)
+        {
+            if (environmentVariableReader != null)
+            {
+                _environmentVariableReader = environmentVariableReader;
+            }
+        }
+
+        // For testing purposes only
+        internal PackageArchiveReader(string filePath, IEnvironmentVariableReader environmentVariableReader)
```

Comment:
do you need both test constructors?
The goal is to test 1 method, so not sure if there's a need for 2 constructors.

Reply:
Ok. I made just 1.